source/request-as-event-emitter.ts

```diff
@@ -12,7 +12,7 @@ import getResponse from './get-response';
 import {normalizeRequestArguments} from './normalize-arguments';
 import {createProgressStream} from './progress';
 import timedOut, {TimeoutError as TimedOutTimeoutError} from './utils/timed-out';
-import {NormalizedOptions, Response, ResponseObject} from './utils/types';
+import {GenericError, NormalizedOptions, Response, ResponseObject} from './utils/types';
```

Comment:
I don't like the name `GenericError` as "generic" means something different with types. Maybe `BaseError` or `GeneralError`?

Reply:
Renamed to `GeneralError`.